libraries/shared/logs/extractor_test.go

```diff
@@ -398,24 +398,53 @@ var _ = Describe("Log extractor", func() {
 				Topic:               fakes.FakeHash.Hex(),
 				StartingBlockNumber: rand.Int63(),
 			}
-			extractor.AddTransformerConfig(fakeConfig)
+			addErr := extractor.AddTransformerConfig(fakeConfig)
+			Expect(addErr).NotTo(HaveOccurred())
 			mockHeaderRepository := &fakes.MockHeaderRepository{}
 			extractor.HeaderRepository = mockHeaderRepository
-			endingBlock := rand.Int63()
+			endingBlock := fakeConfig.StartingBlockNumber + 1
 
 			_ = extractor.BackFillLogs(endingBlock)
 
-			Expect(mockHeaderRepository.GetHeadersInRangeStartingBlock).To(Equal(fakeConfig.StartingBlockNumber))
-			Expect(mockHeaderRepository.GetHeadersInRangeEndingBlock).To(Equal(endingBlock))
+			Expect(mockHeaderRepository.GetHeadersInRangeStartingBlocks).To(ContainElement(fakeConfig.StartingBlockNumber))
+			Expect(mockHeaderRepository.GetHeadersInRangeEndingBlocks).To(ContainElement(endingBlock))
+		})
+
+		It("chunks headers if range greater than interval defined by HeaderChunkSize", func() {
+			fakeConfig := event.TransformerConfig{
+				ContractAddresses:   []string{fakes.FakeAddress.Hex()},
+				Topic:               fakes.FakeHash.Hex(),
+				StartingBlockNumber: rand.Int63(),
+			}
+
+			addErr := extractor.AddTransformerConfig(fakeConfig)
+			Expect(addErr).NotTo(HaveOccurred())
+			mockHeaderRepository := &fakes.MockHeaderRepository{}
+			extractor.HeaderRepository = mockHeaderRepository
+			endingBlock := fakeConfig.StartingBlockNumber + logs.HeaderChunkSize*2
+
+			_ = extractor.BackFillLogs(endingBlock)
+
+			Expect(mockHeaderRepository.GetHeadersInRangeStartingBlocks).To(ConsistOf([]int64{
+				fakeConfig.StartingBlockNumber,
+				fakeConfig.StartingBlockNumber + logs.HeaderChunkSize,
+			}))
+			Expect(mockHeaderRepository.GetHeadersInRangeEndingBlocks).To(ConsistOf([]int64{
+				fakeConfig.StartingBlockNumber + logs.HeaderChunkSize - 1,
+				fakeConfig.StartingBlockNumber + logs.HeaderChunkSize*2 - 1,
+			}))
 		})
 
 		It("returns error if getting headers in range returns error", func() {
 			mockHeaderRepository := &fakes.MockHeaderRepository{}
 			mockHeaderRepository.GetHeadersInRangeError = fakes.FakeError
 			extractor.HeaderRepository = mockHeaderRepository
-			addTransformerConfig(extractor)
+			startingBlock := rand.Int63()
```

Comment:
Just want to make sure i'm reading this right - are these changes to the config required for this specific test? Curious because we're not asserting the starting and ending blocks specifically.

Reply:
We need the `startingBlock` to be sure we call `BackFillLogs` with a number greater than it, but it's a bit tedious to need these 4 lines for every test. Do you think it would make sense to have `addTransformerConfig` (which wraps most of this stuff) return the `startingBlock` it used?

Reply:
Oh gotcha, makes sense we'd need to pass a startingBlock number. Having `addTransformerConfig` return a startingBlock make sense to me!